aya/src/maps/perf/async_perf_event_array.rs

```diff
@@ -196,10 +196,37 @@ impl<T: DerefMut<Target = Map>> AsyncPerfEventArrayBuffer<T> {
     }
 }
 
-impl TryFrom<MapRefMut> for AsyncPerfEventArray<MapRefMut> {
+// Implement try_from()'s here instead of in macro since it's functionality
+// is dependent on the "async" feature flag being enabled.
+impl<'a> TryFrom<&'a Map> for AsyncPerfEventArray<&'a MapData> {
```

Comment:
Why couldn't we use the macros from aya/src/maps/mod.rs here? I love those macros BTW! :heart_eyes:

Reply:
If we do so it breaks builds for the default aya feature set i.e 
```
[features]
default = []
```

Because the `AsyncPerfEventArray` is only available when the `async` feature is used 

```
#[cfg(feature = "async")]
#[cfg_attr(docsrs, doc(cfg(feature = "async")))]
pub use async_perf_event_array::*;
```

And unfortunately something like 

```
impl_try_from_map!(ProgramArray, 
SockMap, 
PerfEventArray, 
StackTraceMap,
#[cfg(feature = "async")]
#[cfg_attr(docsrs, doc(cfg(feature = "async")))]
AsyncPerfEventArray,
);

Doesn't work :(

Reply:
You could call that macro twice maybe?

```rust
impl_try_from_map!(ProgramArray, 
SockMap, 
PerfEventArray, 
StackTraceMap,
):

#[cfg(feature = "async")]
#[cfg_attr(docsrs, doc(cfg(feature = "async")))]
impl_try_from_map!(AsyncPerfEventArray,);
```

Maybe that will work.

Reply:
Hrm that's a good idea! let me give it a try

Reply:
That worked please see recent fixes, thanks!!